File: 프로그래머스/lv1/12931. 자릿수 더하기/자릿수 더하기.js
function solution(n)
{
    let answer = 0;
    answer = parseInt(n);
    console.log(answer);
    
    let arr = answer.toString().split('');
    // console.log(arr);
    // console.log(arr[0]);
    // console.log(arr[1]);
    // console.log(arr[2]);
    // console.log(arr.length);
    
    answer = 0;
    
    for (let i = 0; i < arr.length; i++) {
        answer = answer + parseInt(arr[i]);
    }
    
    return answer;
}

File: 프로그래머스/0/120906. 자릿수 더하기/자릿수 더하기.java
public class Solution {
    public int solution(int n) {
        int answer = 0;
        while (n > 0) {
            answer += n % 10;  // n의 마지막 자릿수를 answer에 더함
            n /= 10;  // n을 10으로 나누어 마지막 자릿수를 제거
        }
        return answer;
    }
}

// class Solution {
//     public int solution(int n) {
//         int answer = 0;
//         String str = Integer.toString(n);
//         String[] arr = str.split("");

//         for(int i = 0; i < arr.length; i++) {
//             answer += Integer.parseInt(arr[i]);
//         }
//         return answer;
//     }
// }
